### PR TITLE
sqlbase: Add system.scheduled_jobs table

### PR DESCRIPTION
**Pull request by miretskiy**
Informs #49346

Add system.scheduled_job table to system catalog.
Add sql migration code to create table when upgrading.

See https://github.com/cockroachdb/cockroach/blob/master/docs/RFCS/20200414_scheduled_jobs.md

Release Notes: None

### CODE REVIEW

File: pkg/sql/sqlbase/system.go [hunk 1]
```diff
@@ -1618,6 +1698,9 @@ func addSystemDescriptorsToSchema(target *MetadataSchema) {
 	target.AddDescriptor(keys.SystemDatabaseID, &StatementBundleChunksTable)
 	target.AddDescriptor(keys.SystemDatabaseID, &StatementDiagnosticsRequestsTable)
 	target.AddDescriptor(keys.SystemDatabaseID, &StatementDiagnosticsTable)
+
+	// Tables introduced in 20.2
```
Comment (pbardea) [inline pkg/sql/sqlbase/system.go:1702]: nit: `.` at the end of this line

**Review by pbardea — APPROVED** [review]
This LGTM. We'll probably want to include this table in the set of system tables included in full cluster backups, but that can be done as a follow up.

File: pkg/clusterversion/cockroach_versions.go [hunk 2]
```diff
@@ -502,10 +503,14 @@ var versionsSingleton = keyedVersions([]keyedVersion{
 	},
 	{
 		// VersionAlterSystemJobsTable is a version which modified system.jobs table
-		//
 		Key:     VersionAlterSystemJobsAddCreatedByColumns,
 		Version: roachpb.Version{Major: 20, Minor: 1, Unstable: 6},
 	},
+	{
+		// VersionAddScheduledJobsTable is a version which adds system.scheduled_jobs table
```
Comment (pbardea) [inline pkg/clusterversion/cockroach_versions.go:510]: (nit: also while we're here, could we add a `.` to the end of this comment and the one above)

**Review by miretskiy — COMMENTED** [review]
I'm working on making CI green.  This basically involves updating bunch of "golden" test files for various tests.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49765#-:-M8pQN2UBE_uWCeidGHK:b-b8f1ie)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @pbardea)

___
*[pkg/clusterversion/cockroach_versions.go, line 510 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49765#-M8l9OdwNd2dEMd-dreU-r1-510:-M8pQBUXAy11sOqNXV5E:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a0570d22b1503ec2605da08d2b592427e8d9bcd3/pkg/clusterversion/cockroach_versions.go#L510)):*
<details><summary><i>Previously, pbardea (Paul Bardea) wrote…</i></summary><blockquote>

(nit: also while we're here, could we add a `.` to the end of this comment and the one above)
</blockquote></details>

Done.

File: pkg/sql/opt/exec/execbuilder/testdata/autocommit [hunk 3]
```diff
@@ -37,7 +37,6 @@ WHERE message     LIKE '%r32: sending batch%'
   AND message NOT LIKE '%PushTxn%'
   AND message NOT LIKE '%QueryTxn%'
 ----
-dist sender send  r32: sending batch 1 CPut, 1 EndTxn to (n1,s1):1
```
Comment (pbardea) [inline pkg/sql/opt/exec/execbuilder/testdata/autocommit:40]: Hm - I don't quite follow why we're seeing differences to this test file?

**Review by pbardea — COMMENTED** [review]
Recent changes look good to me except for the changes in `pkg/sql/opt/exec/execbuilder/testdata/autocommit` - do you know why that test data changed? Might be worth a quick peek from someone on opt?

File: pkg/sql/opt/exec/execbuilder/testdata/autocommit (hunk 3 above)
Comment (miretskiy) [reply pkg/sql/opt/exec/execbuilder/testdata/autocommit:40]: @RaduBerinde  Can you please take a look?

Comment (pbardea) [reply pkg/sql/opt/exec/execbuilder/testdata/autocommit:40]: Synching from our offline discussion - it looks like the query is specifically looking for messages from r32 which likely changed. We'll probably want to figure out which range these messages are on and update the query as well.

**Comment by miretskiy** [comment]
@RaduBerinde 
 Can you take a look at this test file change?  I'm trying to add new system table. 
 Some of the tests that failed had to to be rerun with -rewrite flag to regenerate test data.
A bit worried that about the change to the opt/execbuilder test files.

https://github.com/cockroachdb/cockroach/pull/49765/files/5820bcada88e66735b09f367fe60b3ebb5261865#r434058122

**Review by RaduBerinde — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49765)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @miretskiy, @pbardea, and @RaduBerinde)

___
*[pkg/sql/opt/exec/execbuilder/testdata/autocommit, line 36 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49765#-M8rMvf05_nqSvIUXS4Y:-M8rMvf05_nqSvIUXS4Z:bqdmsne) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d94435d3befb10071e5be71c951447e752e0302/pkg/sql/opt/exec/execbuilder/testdata/autocommit#L36)):*
> ```
> query TT
> SELECT operation, message FROM [SHOW KV TRACE FOR SESSION]
> WHERE message     LIKE '%r32: sending batch%'
> ```

These traces all went away.. We probably need to `s/r32/r33` in the filters.

**Review by RaduBerinde — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49765)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @miretskiy and @pbardea)

___
*[pkg/sql/opt/exec/execbuilder/testdata/autocommit, line 36 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49765#-M8rMvf05_nqSvIUXS4Y:-M8rPrkK2Xnu3XrJhQlr:blrkwry) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d94435d3befb10071e5be71c951447e752e0302/pkg/sql/opt/exec/execbuilder/testdata/autocommit#L36)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

These traces all went away.. We probably need to `s/r32/r33` in the filters.
</blockquote></details>

We should do something like this: https://gist.github.com/RaduBerinde/1808aef02f651f780cf16f4d244fa790

**Review by miretskiy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49765)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @pbardea and @RaduBerinde)

___
*[pkg/sql/opt/exec/execbuilder/testdata/autocommit, line 36 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49765#-M8rMvf05_nqSvIUXS4Y:-M8uNN_W5ImddL1VoGIK:b3g1jeo) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d94435d3befb10071e5be71c951447e752e0302/pkg/sql/opt/exec/execbuilder/testdata/autocommit#L36)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

We should do something like this: https://gist.github.com/RaduBerinde/1808aef02f651f780cf16f4d244fa790
</blockquote></details>

Thanks for the suggestion.  I tried applying your patch.
Alas, it appears that the expected output (dist sender....) cannot be parameterized
Of course, that can be changed, in the logic test, but I'm not that familiar with it
(there is hashing involved, etc).

So, I just replaced r32 w/ r33 for now, and the test passes.

@raduberinde  Please let me know if you think that's acceptible.

**Comment by RaduBerinde** [comment]
___
*[pkg/sql/opt/exec/execbuilder/testdata/autocommit, line 36 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49765#-M8rMvf05_nqSvIUXS4Y:-M8ucUow2eDe4WazGcLD:b-g8w5fa) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d94435d3befb10071e5be71c951447e752e0302/pkg/sql/opt/exec/execbuilder/testdata/autocommit#L36)):*
<details><summary><i>Previously, miretskiy (Yevgeniy Miretskiy) wrote…</i></summary><blockquote>

Thanks for the suggestion.  I tried applying your patch.
Alas, it appears that the expected output (dist sender....) cannot be parameterized
Of course, that can be changed, in the logic test, but I'm not that familiar with it
(there is hashing involved, etc).

So, I just replaced r32 w/ r33 for now, and the test passes.

@raduberinde  Please let me know if you think that's acceptible.
</blockquote></details>

Sure but why not apply that patch? The idea is not that the test would pass next time there'sa change, but that you can -rewrite it without ruining it.

**Review by miretskiy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49765)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @pbardea and @RaduBerinde)

___
*[pkg/sql/opt/exec/execbuilder/testdata/autocommit, line 36 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49765#-M8rMvf05_nqSvIUXS4Y:-M8vAUGl0veeWvoSfWqA:b-cm0snm) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d94435d3befb10071e5be71c951447e752e0302/pkg/sql/opt/exec/execbuilder/testdata/autocommit#L36)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

Sure but why not apply that patch? The idea is not that the test would pass next time there'sa change, but that you can -rewrite it without ruining it.
</blockquote></details>

I see.  I applied your patch, and reran w/ rewrite.

@raduberinde take a (hopefully last) look and lg if you think this change is okay.

**Review by RaduBerinde — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49765#-:-M8vBwgbCLWzvuA7OoUM:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @pbardea and @RaduBerinde)

**Comment by pbardea** [comment]
Just a note that we'll want to eventually want to add this table to the list of system tables which are included in full cluster backups (which is unfortunately just an array in `backup_planning.go` for now). This can be done later in a separate PR, after the table is actually being used.

**Pull request merged**